HW4 - task1.py

```diff
@@ -0,0 +1,22 @@
+"""
+Создайте декоратор, который хранит в себе результаты вызовов функций и
+время получения результата за время запуска программы.
+"""
+
+from time import gmtime
+from time import sleep
+from time import strftime
+
+
+def func_calls_history(func):
+    calls_history = []  # don't use nonlocal in wrapper because list is mutable
+
+    def wrapper(*args, **kwargs):
+        func_call_result = func(*args, **kwargs)
+        calls_history.append({'func': func.__name__,
+                              'end_time': strftime("%Y-%m-%d %T", gmtime()),
+                              'result': func_call_result
+                              })
```

Comment:
There is frequent situation when we have some function\input which return values(more than one), and this function\input is called unknown number of times (like "calls_history" in my program). What is the best type to save such sequences? I used list of dicts, but may be list of lists or list of tuples is better? Or may be it's better to create a class?

Reply:
you solution is OK